### PR TITLE
fix(react-rating): Pass missing Rating's itemLabel prop to its state so RatingItem consumes it from context

### LINKED ISSUE #33360: [Bug]: Rating component itemLabel prop doesn't do anything
### Component

Rating

### Package version

9.54.6

### React version

17.0.2

### Environment

```shell
System:
    OS: Windows 11 10.0.26100
    CPU: (16) x64 Intel(R) Xeon(R) Platinum 8370C CPU @ 2.80GHz
    Memory: 35.47 GB / 63.95 GB
  Browsers:
    Edge: Chromium (130.0.2849.56)
    Internet Explorer: 11.0.26100.1882
```

### Current Behavior

I am trying to update the aria-labels for the RatingItems created by the Rating component. The docs tells me `itemLabel` callback will be used to create the label. but this prop seems to be ignored. Even though I am passing this callback the aria-label is always the rating number. I have put a debugger in the callback and I can tell that it is never getting triggered.

### Expected Behavior

The callback passed to itemLabel should be called for each rating item and aria-label should be created accordingly.

### Reproduction

https://stackblitz.com/edit/ukoutt?file=src%2Fexample.tsx

### Steps to reproduce

Just launch the app.
Use devtools to look at the aria-labels,
Or use screen reader with keyboard navigation, the label that is announced is just the rating number.

### Are you reporting an Accessibility issue?

yes

### Suggested severity

High - No workaround

### Products/sites affected

In app feedback sdk

### Are you willing to submit a PR to fix?

no

### Validations

- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [x] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by nanabear0**
## Previous Behavior

The Rating component didn't use the itemLabel prop, so there wasn't an option to change aria label for the ratin items

## New Behavior

Now it is correctly passing it to the RatingItemContext so aria-labels are created correcty.

## Related Issue(s)

- Fixes #33360

### CODE REVIEW

File: packages/react-components/react-rating/library/src/components/Rating/useRating.tsx [hunk 1]
```diff
@@ -11,6 +11,8 @@ import type { RatingProps, RatingState } from './Rating.types';
 import { RatingItem } from '../../RatingItem';
 import { StarFilled, StarRegular } from '@fluentui/react-icons';
 
+const defaultItemLabel = (num: number) => num + '';
+
```
Comment (sopranopillow) [inline packages/react-components/react-rating/library/src/components/Rating/useRating.tsx:15]: useRatingItem already has a default item label, we only need to pass the item label to the state so the user provided function is consumed.

```suggestion
```

File: packages/react-components/react-rating/library/src/components/Rating/useRating.tsx [hunk 2]
```diff
@@ -31,6 +33,7 @@ export const useRating_unstable = (props: RatingProps, ref: React.Ref<HTMLDivEle
     onChange,
     step = 1,
     size = 'extra-large',
+    itemLabel = defaultItemLabel,
```
Comment (sopranopillow) [inline packages/react-components/react-rating/library/src/components/Rating/useRating.tsx:36]:
```suggestion
    itemLabel,
```

File: change/@fluentui-react-rating-94e96782-4be8-468e-b1a3-0b2ac9a76c62.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "Rating component now correctly uses the itemLabel prop",
```
Comment (sopranopillow) [inline change/@fluentui-react-rating-94e96782-4be8-468e-b1a3-0b2ac9a76c62.json:3]:
```suggestion
  "comment": "fix: Pass missing Rating's itemLabel prop its state so RatingItem consumes it from context.",
```

File: packages/react-components/react-rating/stories/src/Rating/RatingItemLabel.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+import { Rating } from '@fluentui/react-components';
+
+export const ItemLabel = () => {
+  return <Rating itemLabel={num => `item #${num}`} />;
+};
+
+ItemLabel.parameters = {
+  docs: {
+    description: {
+      story: 'You can change the `aria-label` for each rating item using the `itemLabel` prop',
```
Comment (sopranopillow) [inline packages/react-components/react-rating/stories/src/Rating/RatingItemLabel.stories.tsx:11]: I'm not sure if we need a story for this, but I'll leave that up to @microsoft/cxe-prg.

File: packages/react-components/react-rating/library/src/components/Rating/useRating.tsx (hunk 1 above)
Comment (nanabear0) [reply packages/react-components/react-rating/library/src/components/Rating/useRating.tsx]: I was getting an exception when state was created with undefined itemLabel. I guess it was a temporary issue with my build. Applying both of the suggestions.

File: change/@fluentui-react-rating-94e96782-4be8-468e-b1a3-0b2ac9a76c62.json (hunk 3 above)
Comment (nanabear0) [reply change/@fluentui-react-rating-94e96782-4be8-468e-b1a3-0b2ac9a76c62.json]: Thanks for recommending this update, is there a guideline for how these changelogs should be phrased?

File: packages/react-components/react-rating/library/src/components/Rating/Rating.test.tsx [hunk 5]
```diff
@@ -51,4 +51,12 @@ describe('Rating', () => {
     expect(onChange.mock.calls[1][1].value).toBe(3);
     expect(onChange.mock.calls[2][1].value).toBe(2);
   });
+  it('creates RadioItems with correct aria-labels', () => {
+    const onChange = jest.fn();
+    const { getAllByRole } = render(<Rating onChange={onChange} itemLabel={(num: number) => `item #${num}`} />);
```
Comment (dmytrokirpa) [inline packages/react-components/react-rating/library/src/components/Rating/Rating.test.tsx:56]: nit: the prop is already typed, so you could strip the types there.

```suggestion
    const { getAllByRole } = render(<Rating onChange={onChange} itemLabel={(num) => `item #${num}`} />);
```

File: packages/react-components/react-rating/stories/src/Rating/RatingItemLabel.stories.tsx (hunk 4 above)
Comment (dmytrokirpa) [reply packages/react-components/react-rating/stories/src/Rating/RatingItemLabel.stories.tsx:11]: Yeah, I agree, no need to make a separate story for this.

Comment (nanabear0) [reply packages/react-components/react-rating/stories/src/Rating/RatingItemLabel.stories.tsx]: removed the story as suggested

File: packages/react-components/react-rating/library/src/components/Rating/Rating.test.tsx (hunk 5 above)
Comment (nanabear0) [reply packages/react-components/react-rating/library/src/components/Rating/Rating.test.tsx]: Applied suggestion

**Review by sopranopillow — APPROVED** [review]

File: change/@fluentui-react-rating-94e96782-4be8-468e-b1a3-0b2ac9a76c62.json [hunk 6]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-rating-94e96782-4be8-468e-b1a3-0b2ac9a76c62.json:1]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv9 Visual Regression Report](https://aka.ms/visual-changes-review?organization=uifabric&project=fabricpublic&prId=33361&commitId=c648a1c426f4be7820fa4e64e1f423ac374eba3c&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.badgeMask.chromium.png|2| Changed|
|Avatar Converged.Badge Mask RTL.chromium.png|6| Changed|

</details>


  <details><summary><b>Drawer</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Drawer.Full Overlay RTL.chromium.png|1168| Changed|
|Drawer.overlay drawer full.chromium.png|1147| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Pull request merged**